source/agent/analytics/videoGstPipeline/VideoGstAnalyzer.cc

```diff
@@ -272,12 +288,23 @@ void VideoGstAnalyzer::new_sample_from_sink (GstElement * source, gpointer data)
     owt_base::Frame outFrame;
     memset(&outFrame, 0, sizeof(outFrame));
 
-    outFrame.format = owt_base::FRAME_FORMAT_H264;
+    if (pStreamObj->codec.compare("vp8") == 0) {
+        ELOG_INFO("Output video codec is vp8, data size:%d map[0]:%0x:%0x:%0x:%0x:%0x\n", map.size,map.data[0], map.data[2], map.data[3],map.data[4],map.data[5]);
+        outFrame.format = owt_base::FRAME_FORMAT_VP8;
+        if (map.data[3]!=0x9d||map.data[4]!=0x01||map.data[5]!=0x2a) {
```

Comment:
spacing

Reply:
though this is fine， but I think you should check the first bit of the frame header for key frame first. the sync code is supplemental for the check. At the same time, for the key frame it's the right point to get actual width/height for VP8.